tests/test_types.py

```diff
@@ -2002,3 +2002,90 @@ def test_dumping_passes_context_to_inner_type_dump(self):
         assert inner_type.dump_context == context
+
+
+class TestTransform:
+    def test_loading_calls_pre_load_with_original_value(self):
+        class Callbacks():
+            def pre_load(self, data):
+                self.pre_loaded = data
+                return data
+        callbacks = Callbacks()
+        Transform(SpyType(), pre_load=callbacks.pre_load).load('foo')
+        assert callbacks.pre_loaded == 'foo'
+
+    def test_dump_calls_pre_dump_with_original_value(self):
+        class Callbacks():
+            def pre_dump(self, data):
+                self.pre_dump = data
+                return data
+        callbacks = Callbacks()
+        Transform(SpyType(), pre_dump=callbacks.pre_dump).dump('foo')
+        assert callbacks.pre_dump == 'foo'
+
+    def test_loading_calls_inner_type_load_with_result_of_pre_load(self):
+        inner_type = SpyType()
+        transform = Transform(inner_type, pre_load=lambda foo: 'bar')
+        transformed = transform.load('foo')
+        assert inner_type.loaded == 'bar'
+
+    def test_loading_calls_post_load_with_result_of_inner_type_load(self):
+        inner_type = SpyType()
+        transform = Transform(inner_type, post_load=lambda foo: foo + 'bar')
+        transformed = transform.load('foo')
+        assert inner_type.loaded == 'foo'
+        assert transform.load('foo') == 'foobar'
+
+    def test_dumping_calls_inner_type_dump_with_result_of_pre_dump(self):
+        inner_type = SpyType()
+        transform = Transform(inner_type, pre_dump=lambda foo: 'bar')
+        transformed = transform.dump('foo')
+        assert inner_type.dumped == 'bar'
+
+    def test_dumping_calls_post_dump_with_result_of_inner_type_dump(self):
+        inner_type = SpyType()
+        transform = Transform(inner_type, post_dump=lambda foo: foo + 'bar')
+        transformed = transform.dump('foo')
+        assert transform.dump('foo') == 'foobar'
+
+    def test_transform_passes_context_to_inner_type_load(self):
+        inner_type = SpyType()
+        context = object()
+        transform = Transform(inner_type)
+        transform.load('foo', context)
+        assert inner_type.load_context == context
+
+    def test_transform_passes_context_to_inner_type_dump(self):
+        inner_type = SpyType()
+        context = object()
+        transform = Transform(inner_type)
+        transform.dump('foo', context)
+        assert inner_type.dump_context == context
+
+    def test_transform_passes_context_to_pre_load(self):
+        inner_type = SpyType()
+        context = object()
+        transform = Transform(inner_type,
+                              pre_load=lambda foo, context: [foo, context])
+        assert transform.load('foo', context) == ['foo', context]
+
+    def test_transform_passes_context_to_post_load(self):
+        inner_type = SpyType()
+        context = object()
+        transform = Transform(inner_type,
+                              post_load=lambda foo, context:[foo,context])
```

Comment:
Inconsistent spacing: in previous test you had space after colon and space after comma, but in this one there are no spaces. Convention is to put space after colon and comma.